engine/execution/state/state.go

```diff
@@ -341,6 +336,14 @@ func (s *state) SaveExecutionResults(
 		return fmt.Errorf("could not persist execution result: %w", err)
 	}
 
+	// the state commitment is the last data item to be stored, so that
+	// IsBlockExecuted can be implemented by checking whether state commitment exists
+	// in the database
+	err = s.commits.BatchStore(blockID, result.CurrentEndState(), batch)
+	if err != nil {
+		return fmt.Errorf("cannot store state commitment: %w", err)
+	}
+
 	err = batch.Flush()
```

Comment:
does the order actually matter?  they are flush at the same time by batch.Flush?